wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController+ScrollViewDelegate.swift

```diff
@@ -27,6 +27,37 @@ extension ConversationContentViewController: UIScrollViewDelegate {
 
     func scrollViewDidScroll(_ scrollView: UIScrollView) {
         dataSource.didScroll(tableView: scrollView as! UITableView)
+        updateScrollToBottomButtonVisibility()
+    }
+
+    /// Updates the visibility of the `scrollToBottomButton`.
+    ///
+    /// This method checks the `isScrolledToBottom` property to determine whether the button should be hidden or shown.
+    /// If the button's current visibility state is different from what it should be, it either directly sets the visibility
+    /// or performs a fade in/out animation based on the user's accessibility settings for reduced motion.
+    ///
+    /// When "Reduce Motion" is enabled in accessibility settings, the button's visibility is changed without animation
+    /// to respect the user's preference for reduced motion. Otherwise, a 0.25 second fade animation is used for
+    /// a smoother transition between visible and hidden states.
+    ///
+    /// - Note: This method should be called whenever the scroll position changes, typically in response to a
+    ///   `scrollViewDidScroll` event, to ensure that the button's visibility accurately reflects whether the
+    ///   user is scrolled to the bottom of the content.
+    func updateScrollToBottomButtonVisibility() {
+        let shouldHideButton = isScrolledToBottom
+
+        if scrollToBottomButton.isHidden != shouldHideButton {
+            if UIAccessibility.isReduceMotionEnabled {
+                scrollToBottomButton.alpha = shouldHideButton ? 0 : 1
+                scrollToBottomButton.isHidden = shouldHideButton
+            } else {
+                UIView.animate(withDuration: 0.25, animations: {
+                    self.scrollToBottomButton.alpha = shouldHideButton ? 0 : 1
+                }) { _ in
+                    self.scrollToBottomButton.isHidden = shouldHideButton
+                }
+            }
+        }
```

Comment:
Would it also work to set duration 0 if reduced motion is enabled?

Reply:
Didn't try but in terms of responsibilities I wanted to keep those two things separate. Only use the animation block when `UIAccessibility.isReduceMotionEnabled == false`